.github/actions/setup-matrix/action.yml

```
@@ -0,0 +1,23 @@
+name: 'Setup Matrix'
+description: 'Setup a test matrix'
+
+inputs:
+  shards:
```

```suggestion
  shard_count:
```
I'd prefer to see this more explicit in that it is a count, not an actual "shard" array or something.